frontend/components/form/FormHeader.tsx

```diff
@@ -28,47 +28,53 @@ const FormHeader = ({ createMode, state, prevOptionIds }: iFormHeaderProps) => {
   const [, setSuccess] = useLocalStorage<string | null>('success', null)
   const [error, setError] = useState<string | null>(null)
 
-  const onSubmit = async () => {
-    if (
-      !isPost(state) &&
-      (state.options[0]?.choice === '' || state.options[1]?.choice === '')
-    ) {
-      setError('Polls must have at least 2 options')
-      return
+  const getBody = () => {
+    if (!isPost(state)) {
+      return state
     }
-
     const form_data = new FormData()
     if (isPost(state)) {
       Object.entries(state).forEach(([key, value]) => {
         if (key === 'start_date' || key === 'expire_date') {
-          const val = value.toISOString()
+          const val = (value as Date)?.toISOString()
           form_data.append(key, val)
         } else if (key !== 'image') {
-          form_data.append(key, value.toString())
+          form_data.append(key, value?.toString())
         } else {
           form_data.append(key, value)
         }
       })
     }
+    return form_data
+  }
+
+  const onSubmit = async () => {
+    if (
+      !isPost(state) &&
+      (state.options[0]?.choice === '' || state.options[1]?.choice === '')
+    ) {
+      setError('Polls must have at least 2 options')
+      return
+    }
 
     const res = await doApiRequest(`/api/portal/${route}/`, {
       method: 'POST',
-      body: isPost(state) ? form_data : state,
+      body: getBody(),
     })
 
     if (res.ok) {
       // post each poll option if creating a poll
       if (!isPost(state)) {
         const pollRes = await res.json()
-        state.options.map((option) =>
-          doApiRequest('/api/portal/options/', {
+        state.options.forEach(async (option) => {
```

Comment:
probably don't want to be making async calls using a `forEach` since its behavior isn't typical: https://stackoverflow.com/questions/37576685/using-async-await-with-a-foreach-loop.

Reply:
oh u right ill do normal for loop then